force-app/main/default/classes/SOQL.cls

```diff
@@ -288,9 +399,17 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
-
-        // Create a set of email domains from the Leads
-
+        List <Lead> listOfLeads = [SELECT Id, Email, Name, Company, LastName, Status, Website FROM Lead WHERE Id IN :leadIds];
+        // Create a set of email domains from the Leads 
+        Set<String> emailDomains=new Set<String>();
+        for(Lead tempLead: listOfLeads){
+            if(tempLead.Email!=null){
+                String tempLeadDomain = tempLead.Email.split('@')[1];
```

Comment:
Try to use the substringAfter() method

Reply:
Thank you Iwas not aware of that function